go/tasks/plugins/k8s/container/container.go

```diff
@@ -62,8 +62,15 @@ func (Plugin) BuildResource(ctx context.Context, taskCtx pluginsCore.TaskExecuti
 
 	pod := flytek8s.BuildPodWithSpec(podSpec)
 
-	// We want to Also update the serviceAccount to the serviceaccount of the workflow
-	pod.Spec.ServiceAccountName = taskCtx.TaskExecutionMetadata().GetK8sServiceAccount()
+	serviceAccountName := flytek8s.GetServiceAccountNameFromSecurityContext(taskCtx.TaskExecutionMetadata().GetSecurityContext())
```

Comment:
so in the previous case, we have a key-value pair, here it seems like an enum.
I think a cleaner interface would be to always make it enum, but for non supported enums, have a string?

Reply:
@kumare3 The question is how to add enums. In FlyteIDL, any securitycontext stuff would then need to be added everywhere to be plumbed.

Reply:
Chatted offline